crates/serde/src/quantity.rs

```diff
@@ -43,7 +43,7 @@ pub mod opt {
         S: Serializer,
     {
         match value {
-            Some(value) => super::serialize(value, serializer),
+            Some(value) => Some(value.into_ruint()).serialize(serializer),
```

Comment:
this should be

```suggestion
            Some(value) => serializer.serialize_some(value.into_ruint()),
```

see Serialize impl on option which does another match

Reply:
right